pkg/compute/serve.go

```diff
@@ -0,0 +1,287 @@
+package compute
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/blang/semver"
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/filesystem"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/cli/pkg/update"
+)
+
+// ServeCommand produces and runs an artifact from files on the local disk.
+type ServeCommand struct {
+	common.Base
+	manifest         manifest.Data
+	build            *BuildCommand
+	viceroyVersioner update.Versioner
+
+	// Build fields
+	name       common.OptionalString
+	lang       common.OptionalString
+	includeSrc common.OptionalBool
+	force      common.OptionalBool
+
+	// Viceroy fields
+	env common.OptionalString
+}
+
+// NewServeCommand returns a usable command registered under the parent.
+func NewServeCommand(parent common.Registerer, globals *config.Data, build *BuildCommand, viceroyVersioner update.Versioner) *ServeCommand {
+	var c ServeCommand
+
+	c.build = build
+	c.viceroyVersioner = viceroyVersioner
+
+	c.Globals = globals
+	c.CmdClause = parent.Command("serve", "Build and run a Compute@Edge package locally")
+
+	c.manifest.File.SetOutput(c.Globals.Output)
+	c.manifest.File.Read(manifest.Filename)
+
+	// Build flags
+	c.CmdClause.Flag("name", "Package name").Action(c.name.Set).StringVar(&c.name.Value)
+	c.CmdClause.Flag("language", "Language type").Action(c.lang.Set).StringVar(&c.lang.Value)
+	c.CmdClause.Flag("include-source", "Include source code in built package").Action(c.includeSrc.Set).BoolVar(&c.includeSrc.Value)
+	c.CmdClause.Flag("force", "Skip verification steps and force build").Action(c.force.Set).BoolVar(&c.force.Value)
+
+	// Viceroy flags
+	c.CmdClause.Flag("env", "The environment to use when selecting backend definitions (e.g. stage)").Action(c.env.Set).StringVar(&c.env.Value)
```

Comment:
While currently the only object we support is backends we will eventually allow to configure more, so consider re-wording this description. Something like:
```suggestion
	c.CmdClause.Flag("env", "The environment configuration to use (e.g. stage)").Action(c.env.Set).StringVar(&c.env.Value)
```